### PR TITLE
Improve `__all__` definitions for `threading` and `_dummy_threading` modules

### PR DESCRIPTION
**Pull request by AlexWaygood**
Refs https://github.com/PyCQA/flake8-pyi/pull/176

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Thanks, can you tell me what triggered the allowlist changes?

**Comment by AlexWaygood** [comment]
> Thanks, can you tell me what triggered the allowlist changes?

There were a large number of "unused allowlist" complaints on 3.6-3.8 due to adding `__all__` to `_dummy_threading.pyi` (I think these had previously not been imported by `dummy_threading` from `_dummy_threading`, due to the lack of `__all__` in the stub). There were also several new "exists in the stub but not at runtime" complaints for 3.6-3.8 for the `dummy_threading` module -- again, I think this is because stubtest was simply ignoring these classes in the past, because they weren't being imported from `_dummy_threading` into `dummy_threading`.

Ideally we'd fix these problems (and I can do so in this PR if you like), but it felt out of scope for this PR.

File: stdlib/threading.pyi [hunk 1]
```diff
@@ -8,7 +8,84 @@ _TF = Callable[[FrameType, str, Any], Optional[Callable[..., Any]]]
...
+        "BrokenBarrierError",
+        "Timer",
+        "ThreadError",
+        "setprofile",
+        "settrace",
+        "local",
+        "stack_size",
+        "excepthook",
+        "ExceptHookArgs",
+        "gettrace",
```
Comment (JelleZijlstra) [inline stdlib/threading.pyi:36]: I don't see these last two in the code on 3.9.

I do see `get_native_id`, though it looks like it's absent on some OSes: https://docs.python.org/3.10/library/threading.html#threading.get_native_id

Comment (JelleZijlstra) [reply stdlib/threading.pyi:36]: Oops, this is a 3.10 block and gettrace and getprofile were only added in 3.10. Sorry for that.

Comment (AlexWaygood) [reply stdlib/threading.pyi:36]: Good catch on `get_native_id`. I was copy-pasting `__all__` from the source code, and didn't spot that it was conditionally appended to `__all__` lower down.

Comment (AlexWaygood) [reply stdlib/threading.pyi:36]: I have added `get_native_id` to `threading.__all__`, and I have deleted `get_native_id` from `_dummy_threading.pyi`, as stubtest correctly points out that it doesn't exist in `dummy_threading` at runtime.

**Pull request merged**